### PR TITLE
fix: Allow partitions from ingesters to overlap in RPC write mode

### PR DESCRIPTION
**Pull request by carols10cents**
Connects to #6421, [namely this comment](https://github.com/influxdata/influxdb_iox/issues/6421#issuecomment-1361580387).

### CODE REVIEW

File: querier/src/table/mod.rs [hunk 1]
```diff
@@ -442,7 +449,19 @@ impl QuerierTable {
             .map(|sharder| vec![**sharder.shard_for_query(&self.table_name, &self.namespace_name)]);
 
         // get cached table w/o any must-coverage information
-        let Some(cached_table) = self.chunk_adapter.catalog_cache().namespace().get(Arc::clone(&self.namespace_name), &[], span_recorder.child_span("get namespace")).await.and_then(|ns| ns.tables.get(&self.table_name).cloned()) else {return Ok(vec![])};
```
Comment (domodwyer) [inline querier/src/table/mod.rs:445]: WRAP SOME LINES YEAH?

File: querier/src/table/mod.rs [hunk 2]
```diff
@@ -459,19 +478,21 @@ impl QuerierTable {
...
-                    return Err(Error::IngestersOverlap {
-                        ingester1: Arc::clone(o.get()),
-                        ingester2: Arc::clone(partition.ingester()),
-                        partition: partition.partition_id(),
-                    })
-                }
-                Entry::Vacant(v) => {
-                    v.insert(Arc::clone(partition.ingester()));
+        if !self.rpc_write() {
+            // check that partitions from ingesters don't overlap
```
Comment (domodwyer) [inline querier/src/table/mod.rs:482]: I wonder if this comment should have a "because if they do, X will explode when Y happens" type comment...

**Review by domodwyer — APPROVED** [review]

File: querier/src/table/mod.rs (hunk 1 above)
Comment (carols10cents) [reply querier/src/table/mod.rs:445]: Yeah I don't think rustfmt understands let/else yet :(

File: querier/src/table/mod.rs (hunk 2 above)
Comment (carols10cents) [reply querier/src/table/mod.rs:482]: Shoulda coulda woulda, but it should get deleted soon.... 🙏🏻

**Pull request merged**